src/client_networking.rs

```diff
@@ -55,49 +55,23 @@ impl From<StrictError> for ConnectionError {
 }
 
 
-// I'd change the declaration to: request_table(table_name: &str, server_address: &str)
-// Agree with name => table_name but this gets a csv. Should be called download_csv, though, to be consistent with server()
 pub fn request_csv(name: &str, address: &str) -> Result<StrictTable, ConnectionError> {
-    // Validating the server_address somehow and providing a descriptive error would probably be a good idea
 
     let mut connection: TcpStream = TcpStream::connect(address)?;
-    
-    // The writing to the connection here is the same code as elsewhere
-    // perhaps you want to encapsulate the writing/communications themselves somehow. 
-    // E.g. say you change the server to accept json or another format, 
-    // you'd only want to make the formatting change in one place.
-    // It doesn't have to be a class but some intermideary between this request_table function
-    // and the connection that would surface say at least two methods: 
-    // send_request(method: requesting|sending|other, params: &str), 
-    //      where the method param would be an enum allowing you to 
-    //      switch between the appearent methods however that's best done in rust
-    //      and the other would be the params (like name) currently going into the formatted string.
-    //      This method could then also handle parsing the data from the server  (i.e. the read in the loop below)
-    //      and casting the bytes to a string
-    // and 
-    // send_confirmation(), which would send the "OK"
-
-    //      This is this way for the print statement for debugging reasons. Will be this in future:
-    //        connection.write(format!("admin|admin|Requesting|{}", name).as_bytes())?;
```

Comment:
The first comment group was mine and the second was yours.

Okay, so I get your point with it just being a single line once you refactor away the debugging. However, what I was trying to get at is relating to singular responsibility and levels of abstraction. Right now, this function includes information about the communication layer, i.e. the formatting of the string `admin|admin` which I presume is authentication, and even there it has knowledge about how to format the message with the pipe separator.  Then it's specifying the action to perform `Requesting` (which is defined in string here, which may cause misspellings and be difficult to catch if you decide to change the string from `Requesting` to `request` or `get_table` or whatever) and the final `name` parameter. To me these three things might be performed differently later on, esp. Auth but also the structure of the message may change. So I was suggesting an implementation layer between the function we're in here and the actual formatting of the communications. 

Do you follow my line of thought? Do you disagree somehow?

Reply:
And in addition, I forgot about the actual reading (done in the loop below), which you mentioned yourself had been improved in the server but not here, suggesting that it's a similar action which would benefit from using the same utility somehow 🤗

Reply:
Yeah the whole client library is a mess of "will it work this way?" trial and error.

That being said, the auth info will for sure not be hard-coded in the future. It will be an argument and will be hashed before being sent. I need to send the server some auth_info as text so it can authenticate the client. Do you have other suggestions than sending it as text (hashed and encrypted, of course)?

Reply:
In regards to the second comment, yeah, for sure the logic in server will be copied to here, as a common function if I can make that work (not always straightforward because of the borrow checker).

Reply:
> Yeah the whole client library is a mess of "will it work this way?" trial and error.
😅 yeah, that's how these things tend to go 


> That being said, the auth info will for sure not be hard-coded in the future. It will be an argument and will be hashed before being sent. I need to send the server some auth_info as text so it can authenticate the client. Do you have other suggestions than sending it as text (hashed and encrypted, of course)?

No not really, apart from maybe some whitelisting on the server side there's not much else that comes to mind. And regarding how you encrypt here and decode there I'm not really sure at such a low level 😅 that'll be a fun thing to figure out. I'm mostly familiar with the usage of stuff at a higher level of abstraction like OAuth2 which is worth checking out as its industry standard for authed comms.

Reply:
Fuck Oauth, we do it live!!